### ISSUE TITLE
Next Step Bot

### ISSUE THREAD

**Issue by mekarpeles** [internetarchive/openlibrary#10229]
### Proposal

TL;DR before assigning issues to contributors who may be likely to bounce, add one step of useful friction / engagement before assignment that requires the contributor to put skin in the game, engage with a useful prompt, and help us improve the issue or identify questions first.

Issues on Open Library github get assigned to contributors and then become stale and it’s hard to know when they’re stale. One solution is to engage more with contributors prior to assigning them. We hypothesize some of this process may be automated.

Furthermore, by automating some of this process, we may be able to get contributors to take actionable next steps which help us:
1. have confidence they understand the issue (and have skin in the game)
2. identify / understand where they may be stuck
3. enlist the contributor to at least participate in breakdown (to the extent they're able)

This might mean:
* proposing designs for feedback
* identifying (or asking) which files are relevant to the project
* proposing or explaining their approach to addressing the issue (for feedback)
* asking the contributor if they have any question or are blocked by anything

### Justification

**Why** do contributors ask to be assigned to issues and then not follow through?

Contributors may forget or get busy... For these cases, we may be able to automatically add the https://github.com/internetarchive/openlibrary/labels/Needs%3A%20Review%20Assignee label to these issues (or exemptions) as per:
* #9657

In my experience, a more common reason is that a contributor stalls because they may be expecting a mentor to help guide them step-by-step or provide additional guidance.

**Why do they need additional guidance?**

Because the issue may be complicated, overwhelming, or require further  breakdown, and it may be difficult to know what question to even ask in order to get started.

**Why are next steps unclear?**

Because staff sometimes doesn’t provide sufficient breakdown for issues and often times no direction is communicated to the contributor to help them know what followup question to ask.

### Breakdown

When someone asks to be assigned to an issue, depending on the format, have **Next Step Bot** respond to the patron with a followup to make sure they can succeed once assigned.

**Comment by mekarpeles** [comment]
I think this is broadly achieved by the Issue Refinement... Although it is related to #13160 "Needs Response" (and I'd want that link maintained). Work can be continued on that issue.
